mobile/src/profile/home-profile-screen.tsx

```diff
@@ -50,24 +58,39 @@ class ProfileScreenImpl extends Component<ProfileScreenProps & AuthDispatchProps
 
     //----------------------------------------------------------------------------
     public shouldComponentUpdate(nextProps: ProfileScreenProps & AuthDispatchProps) {
-        const employees = this.props.employees.employeesById;
-        const nextEmployees = nextProps.employees.employeesById;
         const requests = this.props.requests;
         const nextRequests = nextProps.requests;
-
         if (!requests.equals(nextRequests)) {
             const calendarEvents = requests.get(this.props.employee.employeeId);
             const nextCalendarEvents = nextRequests.get(nextProps.employee.employeeId);
 
-            return calendarEvents !== nextCalendarEvents;
+            if (calendarEvents !== nextCalendarEvents) {
+                return true;
+            }
         }
 
+        const employees = this.props.employees.employeesById;
+        const nextEmployees = nextProps.employees.employeesById;
         if (!employees.equals(nextEmployees)) {
-            const newEmployeesSubset = nextEmployees.filter(employee => {
-                return !employees.has(employee.employeeId);
+            const newEmployeesSubset = nextEmployees.filter(element => {
+                return !employees.has(element.employeeId);
```

Comment:
Shouldn't it be like `if (!employees.has(element.employeeId)) { return true; }`?

Reply:
@antonm76 no, this is a filter predicate return.